petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -42,16 +46,24 @@ def test_primitive(self):
                 # Now we only have one batch.
             for i in range(converter.dataset_size):
                 for col in df.schema.names:
-                    self.assertEquals(getattr(ts, col)[i], expected_df[i][col])
+                    actual_ele = getattr(ts, col)[i]
+                    expected_ele = expected_df[i][col]
+                    if type(actual_ele) == bytes:
+                        actual_ele = actual_ele.decode()
+                    if type(expected_ele) == bytearray:
+                        expected_ele = expected_ele.decode()
```

Comment:
I think the df.collect will always return "bytes" types, so we can remove this branch code.

Reply:
oh, I am wrong, spark always generate results as "bytearray" type. So remove the "bytes" branch.

Reply:
The bytes branch is for `actual_ele`, not `expected_ele`.